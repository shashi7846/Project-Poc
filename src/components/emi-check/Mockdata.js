export const mockLoginData = {
  PropertyPrice: "10000000",
  LoanAmount: "100000",
  netIncome:"80000",
};
