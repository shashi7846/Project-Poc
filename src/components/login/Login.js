import React, { useState } from "react";
import { Link, useNavigate } from "react-router-dom";
//import { faMailBulk } from "@react-icons/all-files/fa/faMailBulk";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faKey, faMailBulk } from "@fortawesome/free-solid-svg-icons";
// import { GetuserbyEmailAndPassword, Postlogin } from "../../api/Api";

// import { toast } from "react-toastify";
import { useDispatch } from "react-redux";
import { LoginAction } from "../../redux/actions/auth";

export const ValidateLoginForm = (input) => {
  const errors = [];

  console.log(input);
  if (!input.email) {
    errors.push("Email is Required");
  }
  if (!input.password) {
    errors.push("Please enter Password");
  }
  return errors;
};

const LoginData = {
  email: "",
  password: "",
};

function Login() {
  const [Logindetails, setLogindetails] = useState(LoginData);

  const handleChange = (e) =>
    setLogindetails({ ...Logindetails, [e.target.name]: e.target.value });

  let navigate = useNavigate();

  let dispatch = useDispatch();

  const handleLogin = (e) => {
    e.preventDefault();

    dispatch(LoginAction(Logindetails, navigate));
  };

  return (
    <>
      <div className="container mt-5 h-100">
        <div className="d-flex justify-content-center h-100">
          <div className="card col-5" style={{ backgroundColor: "black" }}>
            <div className="card-header Cardtop ">
              <h3 className="login-name text-light">
                <b>Login</b>
              </h3>
            </div>
            <div className="card-body">
              <form onSubmit={handleLogin}>
                <div className="input-group form-group">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faMailBulk}
                        style={{
                          fontSize: "1.75em",
                          color: "black",
                        }}
                      ></FontAwesomeIcon>
                      <label className="text-dark ms-2" for="EmailId">
                        <b>Email Id</b>
                      </label>
                    </span>
                  </div>
                  <input
                    type="email"
                    name="email"
                    aria-label="email"
                    className="form-control ms-2"
                    placeholder="email"
                    value={Logindetails.email}
                    onChange={handleChange}
                    required
                  />
                </div>
                <div className="input-group form-group mt-3">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white "
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faKey}
                        style={{ fontSize: "1.75em", color: "black" }}
                      ></FontAwesomeIcon>
                      <label className="text-dark ms-2">
                        <b>Password</b>
                      </label>
                    </span>
                  </div>
                  <input
                    type="password"
                    name="password"
                    aria-label="password"
                    className="form-control ms-2"
                    placeholder="password"
                    value={Logindetails.password}
                    onChange={handleChange}
                    required
                  />
                </div>

                <div className="form-group">
                  <button
                    type="submit"
                    value="Login"
                    name="Login"
                    aria-label="Login"
                    className="btn float-right  btn-dark mt-2 px-4"
                  >
                    Login
                  </button>
                </div>
              </form>
            </div>
            <div className="card-footer">
              <div className="d-flex justify-content-center links text-light">
                Don't have an account?<Link to="/register">Register</Link>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
}

export default Login;
