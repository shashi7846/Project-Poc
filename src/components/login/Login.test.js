import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderComponent, renderComponent1 } from "../../utils/test_helpers";
import Login from "./Login";
import { mockLoginData } from "./Mockdata";

const user = userEvent.setup();

describe("Login page", () => {
  test("should render initial login page", () => {
    render(renderComponent(<Login />));
    expect(screen.getByRole("heading", { name: "Login" })).toBeInTheDocument();

    expect(screen.getByRole("textbox", { name: "email" })).toBeInTheDocument();

    expect(screen.getByRole("button", { name: "Login" })).toBeInTheDocument();

    expect(screen.getByRole("link", { name: "Register" })).toBeInTheDocument();

    expect(screen.getByText("Email Id")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("password")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("email")).toBeInTheDocument();
  });

  test("should handle input changes in Login", async () => {
    render(renderComponent(<Login />));
    const email = screen.getByRole("textbox", { name: "email" });
    const password = screen.getByPlaceholderText("password");
    await user.type(email, mockLoginData.email);
    expect(email).toHaveValue(mockLoginData.email);

    await user.type(password, mockLoginData.password);
    expect(password).toHaveAttribute("type", "password");
    expect(password).toHaveValue(mockLoginData.password);
  });
  test("msw testing", async () => {
    let { store } = renderComponent1(<Login />);
    const email = screen.getByRole("textbox", { name: "email" });
    const password = screen.getByPlaceholderText("password");
    const button = screen.getByRole("button", { name: "Login" });
    await user.type(email, mockLoginData.email);
    //expect(email).toHaveValue(mockLoginData.email);

    await user.type(password, mockLoginData.password);
    // expect(password).toHaveAttribute("type", "password");
    //expect(password).toHaveValue(mockLoginData.password);

    await user.click(button);
    waitFor(() => {
      expect(store.getState().state.message).toBe("allow");
    });
  });

  test("should handle errors in Login", () => {
    render(renderComponent(<Login />));
    screen.logTestingPlaygroundURL();
    const email = screen.getByRole("textbox", { name: "email" });
    // const password = screen.getByPlaceholderText("password");

    user.click(
      screen.getByRole("button", {
        name: /login/i,
      })
    );
    let error = screen.findByText("Please fill in this field.");
    waitFor(() => {
      expect(error).toBeInTheDocument();
    });

    // password.focus();
    // await user.tab();
    // expect(screen.getByText("Password Required")).toBeInTheDocument();
  });

  test("Should Show Login Heading", async () => {
    render(renderComponent(<Login />));
    // const Heading = screen.getAllByText(/Login/i);
    // expect(Heading[0]).toBeInTheDocument();

    expect(screen.getByRole("heading", { name: "Login" })).toBeInTheDocument();
  });

  test("Matching the snapshot of the Login", async () => {
    const { asFragment } = render(renderComponent(<Login />));
    expect(asFragment()).toMatchSnapshot();
  });
});
