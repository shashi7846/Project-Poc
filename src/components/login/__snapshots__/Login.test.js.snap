// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page Matching the snapshot of the Login 1`] = `
<DocumentFragment>
  <div
    class="container mt-5 h-100"
  >
    <div
      class="d-flex justify-content-center h-100"
    >
      <div
        class="card col-5"
        style="background-color: black;"
      >
        <div
          class="card-header Cardtop "
        >
          <h3
            class="login-name text-light"
          >
            <b>
              Login
            </b>
          </h3>
        </div>
        <div
          class="card-body"
        >
          <form>
            <div
              class="input-group form-group"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text text-white"
                  style="min-width: 150px;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-envelopes-bulk fa-beat-fade"
                    data-icon="envelopes-bulk"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1.75em; color: black;"
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M96 0C78.3 0 64 14.3 64 32V224h96V192c0-35.3 28.7-64 64-64H448V32c0-17.7-14.3-32-32-32H96zM224 160c-17.7 0-32 14.3-32 32v32h96c35.3 0 64 28.7 64 64V416H544c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32H224zm240 64h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H464c-8.8 0-16-7.2-16-16V240c0-8.8 7.2-16 16-16zM32 256c-17.7 0-32 14.3-32 32v13L155.1 415.9c1.4 1 3.1 1.6 4.9 1.6s3.5-.6 4.9-1.6L320 301V288c0-17.7-14.3-32-32-32H32zm288 84.8L184 441.6c-6.9 5.1-15.3 7.9-24 7.9s-17-2.8-24-7.9L0 340.8V480c0 17.7 14.3 32 32 32H288c17.7 0 32-14.3 32-32V340.8z"
                      fill="currentColor"
                    />
                  </svg>
                  <label
                    class="text-dark ms-2"
                    for="EmailId"
                  >
                    <b>
                      Email Id
                    </b>
                  </label>
                </span>
              </div>
              <input
                aria-label="email"
                class="form-control ms-2"
                name="email"
                placeholder="email"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="input-group form-group mt-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text text-white "
                  style="min-width: 150px;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-key fa-beat-fade"
                    data-icon="key"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1.75em; color: black;"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M336 352c97.2 0 176-78.8 176-176S433.2 0 336 0S160 78.8 160 176c0 18.7 2.9 36.8 8.3 53.7L7 391c-4.5 4.5-7 10.6-7 17v80c0 13.3 10.7 24 24 24h80c13.3 0 24-10.7 24-24V448h40c13.3 0 24-10.7 24-24V384h40c6.4 0 12.5-2.5 17-7l33.3-33.3c16.9 5.4 35 8.3 53.7 8.3zm40-176c-22.1 0-40-17.9-40-40s17.9-40 40-40s40 17.9 40 40s-17.9 40-40 40z"
                      fill="currentColor"
                    />
                  </svg>
                  <label
                    class="text-dark ms-2"
                  >
                    <b>
                      Password
                    </b>
                  </label>
                </span>
              </div>
              <input
                aria-label="password"
                class="form-control ms-2"
                name="password"
                placeholder="password"
                required=""
                type="password"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <button
                aria-label="Login"
                class="btn float-right  btn-dark mt-2 px-4"
                name="Login"
                type="submit"
                value="Login"
              >
                Login
              </button>
            </div>
          </form>
        </div>
        <div
          class="card-footer"
        >
          <div
            class="d-flex justify-content-center links text-light"
          >
            Don't have an account?
            <a
              href="/register"
            >
              Register
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Login page Matching the snapshot of the navbar 1`] = `
<DocumentFragment>
  <div
    class="container mt-5 h-100"
  >
    <div
      class="d-flex justify-content-center h-100"
    >
      <div
        class="card col-5"
        style="background-color: black;"
      >
        <div
          class="card-header Cardtop "
        >
          <h3
            class="login-name text-light"
          >
            <b>
              Login
            </b>
          </h3>
        </div>
        <div
          class="card-body"
        >
          <form>
            <div
              class="input-group form-group"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text text-white"
                  style="min-width: 150px;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-envelopes-bulk fa-beat-fade"
                    data-icon="envelopes-bulk"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1.75em; color: black;"
                    viewBox="0 0 576 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M96 0C78.3 0 64 14.3 64 32V224h96V192c0-35.3 28.7-64 64-64H448V32c0-17.7-14.3-32-32-32H96zM224 160c-17.7 0-32 14.3-32 32v32h96c35.3 0 64 28.7 64 64V416H544c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32H224zm240 64h32c8.8 0 16 7.2 16 16v32c0 8.8-7.2 16-16 16H464c-8.8 0-16-7.2-16-16V240c0-8.8 7.2-16 16-16zM32 256c-17.7 0-32 14.3-32 32v13L155.1 415.9c1.4 1 3.1 1.6 4.9 1.6s3.5-.6 4.9-1.6L320 301V288c0-17.7-14.3-32-32-32H32zm288 84.8L184 441.6c-6.9 5.1-15.3 7.9-24 7.9s-17-2.8-24-7.9L0 340.8V480c0 17.7 14.3 32 32 32H288c17.7 0 32-14.3 32-32V340.8z"
                      fill="currentColor"
                    />
                  </svg>
                  <label
                    class="text-dark ms-2"
                    for="EmailId"
                  >
                    <b>
                      Email Id
                    </b>
                  </label>
                </span>
              </div>
              <input
                aria-label="email"
                class="form-control ms-2"
                name="email"
                placeholder="email"
                required=""
                type="email"
                value=""
              />
            </div>
            <div
              class="input-group form-group mt-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text text-white "
                  style="min-width: 150px;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-key fa-beat-fade"
                    data-icon="key"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1.75em; color: black;"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M336 352c97.2 0 176-78.8 176-176S433.2 0 336 0S160 78.8 160 176c0 18.7 2.9 36.8 8.3 53.7L7 391c-4.5 4.5-7 10.6-7 17v80c0 13.3 10.7 24 24 24h80c13.3 0 24-10.7 24-24V448h40c13.3 0 24-10.7 24-24V384h40c6.4 0 12.5-2.5 17-7l33.3-33.3c16.9 5.4 35 8.3 53.7 8.3zm40-176c-22.1 0-40-17.9-40-40s17.9-40 40-40s40 17.9 40 40s-17.9 40-40 40z"
                      fill="currentColor"
                    />
                  </svg>
                  <label
                    class="text-dark ms-2"
                  >
                    <b>
                      Password
                    </b>
                  </label>
                </span>
              </div>
              <input
                aria-label="password"
                class="form-control ms-2"
                name="password"
                placeholder="password"
                required=""
                type="password"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <button
                class="btn float-right  btn-dark mt-2 px-4"
                type="submit"
                value="Login"
              >
                Login
              </button>
            </div>
          </form>
        </div>
        <div
          class="card-footer"
        >
          <div
            class="d-flex justify-content-center links text-light"
          >
            Don't have an account?
            <a
              href="/register"
            >
              Register
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
